### ISSUE TITLE
Issue while training classifier for multiple labels

### ISSUE THREAD

**Issue by premrajnarkhede** [zalandoresearch/flair#358]
Hi All

Thanks for developing this powerful library.

I am using Flair to train model to classify tweet texts into acquisitions, financing, partnerships, licensing and none. (There might be more labels but they hardly have any data)

I am using similar code as given in helper doc just changed multi_label to True

```
from flair.models import TextClassifier
from flair.data import Sentence
from flair.data_fetcher import NLPTaskDataFetcher
from flair.embeddings import WordEmbeddings, FlairEmbeddings, DocumentLSTMEmbeddings
from flair.models import TextClassifier
from flair.trainers import ModelTrainer
from pathlib import Path
corpus = NLPTaskDataFetcher.load_classification_corpus(Path('../InsightsExperiments'), train_file='xaa', test_file='xab', dev_file='xab') 
word_embeddings = [WordEmbeddings('glove'), FlairEmbeddings('news-forward-fast'), FlairEmbeddings('news-backward-fast')]
document_embeddings = DocumentLSTMEmbeddings(word_embeddings, hidden_size=512, reproject_words=True, reproject_words_dimension=256)
classifier = TextClassifier(document_embeddings, label_dictionary=corpus.make_label_dictionary(), multi_label=True)
trainer = ModelTrainer(classifier, corpus)
trainer.train('./', max_epochs=2)
```

I am getting following output
```
2019-01-03 19:50:12,669 ----------------------------------------------------------------------------------------------------
2019-01-03 19:50:12,672 Evaluation method: MICRO_F1_SCORE
2019-01-03 19:50:12,674 ----------------------------------------------------------------------------------------------------
2019-01-03 19:50:13,395 epoch 1 - iter 0/24 - loss 0.02129710
2019-01-03 19:50:14,772 epoch 1 - iter 2/24 - loss 0.01979518
2019-01-03 19:50:16,142 epoch 1 - iter 4/24 - loss 0.01816955
2019-01-03 19:50:17,483 epoch 1 - iter 6/24 - loss 0.01702432
2019-01-03 19:50:18,870 epoch 1 - iter 8/24 - loss 0.01604336
2019-01-03 19:50:20,134 epoch 1 - iter 10/24 - loss 0.01520755
2019-01-03 19:50:21,399 epoch 1 - iter 12/24 - loss 0.01450009
2019-01-03 19:50:22,690 epoch 1 - iter 14/24 - loss 0.01395708
2019-01-03 19:50:24,005 epoch 1 - iter 16/24 - loss 0.01347029
2019-01-03 19:50:25,134 epoch 1 - iter 18/24 - loss 0.01305109
2019-01-03 19:50:26,468 epoch 1 - iter 20/24 - loss 0.01275499
2019-01-03 19:50:27,671 epoch 1 - iter 22/24 - loss 0.01247760
2019-01-03 19:50:28,165 ----------------------------------------------------------------------------------------------------
2019-01-03 19:50:28,167 EPOCH 1 done: loss 0.0124 - lr 0.1000 - bad epochs 0
2019-01-03 19:50:29,834 DEV  : loss 0.00976349 - f-score 0.0000 - acc 0.0000
2019-01-03 19:50:31,434 TEST : loss 0.00976349 - f-score 0.0000 - acc 0.0000
2019-01-03 19:50:36,133 ----------------------------------------------------------------------------------------------------
2019-01-03 19:50:36,726 epoch 2 - iter 0/24 - loss 0.00917903
2019-01-03 19:50:37,950 epoch 2 - iter 2/24 - loss 0.00914586
2019-01-03 19:50:38,994 epoch 2 - iter 4/24 - loss 0.00919277
2019-01-03 19:50:40,212 epoch 2 - iter 6/24 - loss 0.00926150
2019-01-03 19:50:41,488 epoch 2 - iter 8/24 - loss 0.00932142
2019-01-03 19:50:42,878 epoch 2 - iter 10/24 - loss 0.00929928
2019-01-03 19:50:43,988 epoch 2 - iter 12/24 - loss 0.00923532
2019-01-03 19:50:45,060 epoch 2 - iter 14/24 - loss 0.00922431
2019-01-03 19:50:46,230 epoch 2 - iter 16/24 - loss 0.00923761
2019-01-03 19:50:47,182 epoch 2 - iter 18/24 - loss 0.00919971
2019-01-03 19:50:48,079 epoch 2 - iter 20/24 - loss 0.00921001
2019-01-03 19:50:49,313 epoch 2 - iter 22/24 - loss 0.00917710
2019-01-03 19:50:49,851 ----------------------------------------------------------------------------------------------------
2019-01-03 19:50:49,853 EPOCH 2 done: loss 0.0092 - lr 0.1000 - bad epochs 0
2019-01-03 19:50:51,841 DEV  : loss 0.00938275 - f-score 0.0000 - acc 0.0000
2019-01-03 19:50:54,057 TEST : loss 0.00938275 - f-score 0.0000 - acc 0.0000
2019-01-03 19:51:03,398 ----------------------------------------------------------------------------------------------------
2019-01-03 19:51:03,399 Testing using best model ...
2019-01-03 19:51:06,282 MICRO_AVG: acc 0.0 - f1-score 0.0
2019-01-03 19:51:06,284 MARCO_AVG: acc 0.0 - f1-score 0.0
2019-01-03 19:51:06,285 Acquisition tp: 0 - fp: 0 - fn: 99 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,286 Financing  tp: 0 - fp: 0 - fn: 57 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,287 Licensing  tp: 0 - fp: 0 - fn: 24 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,287 None       tp: 0 - fp: 0 - fn: 73 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,288 Operations tp: 0 - fp: 0 - fn: 1 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,289 Opportunities tp: 0 - fp: 0 - fn: 2 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,290 Partnership tp: 0 - fp: 0 - fn: 70 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,291 RetailLocation tp: 0 - fp: 0 - fn: 2 - tn: 0 - precision: 0.0000 - recall: 0.0000 - accuracy: 0.0000 - f1-score: 0.0000
2019-01-03 19:51:06,292 ----------------------------------------------------------------------------------------------------
{'test_score': 0.0,
 'dev_score_history': [0.0, 0.0],
 'train_loss_history': [0.01239741429764563, 0.009227615571458927],
 'dev_loss_history': [0.009763485752046108, 0.009382746182382107]}
```

Model isn't learning anything. What am I doing wrong?

Files used in example are here
xaa
https://ufile.io/z8lu2
xab
https://ufile.io/e5i1p

**Comment by premrajnarkhede** [comment]
Also I use same files for FastText and it trains fine

**Comment by premrajnarkhede** [comment]
@alanakbik 

Thank you your guidance. I have above 90% figures for both recall and preision using Flair on Dev testsets. I am creating a separate test set for better evaluation. Will keep you posted.

Premraj

**Comment by alanakbik** [comment]
hi @premrajnarkhede hope everything is working! I'm closing the issue but feel free to reopen if you have more questions!

**Comment by silversword05** [comment]
TextClassifier is not working properly when loaded from a model file
e.g. classifier = TextClassifier.load('data_files/flair_custom_multi_label_sir/flair-model-details/best-model.pt')
Either providing only one label or none at all.
